### DISCUSSION TITLE
FakePlayer PlayerSkin Incorrect

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#834] [Help]
Hi,

So I'm experimenting with FakePlayers and I have the following code:
```java
        FakePlayerOption option = new FakePlayerOption();
        option.setInTabList(false);
        option.setRegistered(false);

        PlayerSkin playerSkin = PlayerSkin.fromUuid("f7c77d99-9f15-4a66-a87d-c4a51ef30d19");

        FakePlayer.initPlayer(
                UUID.fromString("f498513c-e8c8-4773-be26-ecfc7ed5185d"),
                username,
                option,
                (FakePlayer p) -> {
                    p.setSkin(playerSkin);
                }

        );
```

And upon inspecting the PlayerSkin, it seems that it's returning the correct Base64 data, however, when I log onto the server, the skins displayed are all steves. Is there any reason for this?

EDIT:

It seems that this only happens when the FakePlayer is not registered in the tablist

**Comment by TheMode** [comment]
Complete code would be great to have (thinking of the option).

I would additionally recommend using `PlayerSkinInitEvent`.

**Reply by para-dise** [reply]
I updated my code, and also PlayerSkinInitEvent doesn't seem to be getting fired for FakePlayers, only for real players
